plugins/blockly/blocks/util/apply_first_match.ts

```diff
@@ -2,29 +2,126 @@ import Blockly from "blockly";
 
 Blockly.Blocks["apply_first_match"] = {
   init: function () {
-    this.appendDummyInput().appendField("Apply first match");
-    this.appendValueInput("ACTIONS_ONE").setCheck("Action[]");
-    this.appendValueInput("ACTIONS_TWO").setCheck("Action[]");
-    this.setOutput(true, "Action[]");
+    this.appendDummyInput().appendField("apply first match");
+    this.appendValueInput("ACTIONS0").setCheck("Action[]");
+    this.setPreviousStatement(true);
+    this.setNextStatement(true);
     this.setColour(230);
     this.setTooltip("");
     this.setHelpUrl("");
+
+    this.setMutator(new Blockly.Mutator(["additional_action"]));
+    this.addtional_num_actions = 0;
+  },
+  mutationToDom: function () {
+    if (!this.addtional_num_actions) {
+      return null;
+    }
+    const container = document.createElement("mutation");
+    container.setAttribute("actions", this.addtional_num_actions);
+    return container;
+  },
+  domToMutation: function (xmlElement: HTMLElement) {
+    this.addtional_num_actions =
+      parseInt(xmlElement.getAttribute("actions"), 10) || 0;
+    for (let i = 1; i <= this.addtional_num_actions; i++) {
+      this.appendValueInput("ACTIONS" + i).setCheck("Action[]");
+    }
+  },
+  decompose: function (workspace: Blockly.Workspace) {
+    const containerBlock = workspace.newBlock(
+      "action_container",
+    ) as Blockly.BlockSvg;
+    containerBlock.initSvg();
+
+    let connection = containerBlock.getInput("STACK").connection;
```

Comment:
I think it could be replaced with `const`
```suggestion
    const connection = containerBlock.getInput("STACK").connection;
```